File: simple-manage/src/main/java/org/shanling/simplemanage/entity/ScriptCardDevice.java
package org.shanling.simplemanage.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 卡密关联设备实体类
 * 
 * @author shanling
 */
@Data
@TableName("script_card_device")
public class ScriptCardDevice {

    /**
     * 设备ID（主键，雪花算法生成）
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 卡密
     */
    private String cardNo;

    /**
     * 设备的Android ID
     */
    private String deviceAndroidId;

    /**
     * 屏幕宽度
     */
    private Integer deviceWidth;

    /**
     * 屏幕高度
     */
    private Integer deviceHeight;

    /**
     * 厂商品牌
     */
    private String deviceBrand;

    /**
     * 设备名称
     */
    private String deviceName;

    /**
     * 设备型号
     */
    private String deviceModel;

    /**
     * API版本
     */
    private String deviceSdkInt;

    /**
     * 设备IMEI
     */
    private String deviceImei;

    /**
     * 主板型号
     */
    private String deviceBroad;

    /**
     * 版本号
     */
    private String deviceBuildId;

    /**
     * 备注
     */
    private String remark;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 创建人
     */
    private String createBy;

    /**
     * 更新人
     */
    private String updateBy;

}

File: manage-vue/src/api/scriptCardDevice.ts
import request from '@/utils/request'

export interface ScriptCardDevice {
  id?: string
  cardNo: string
  deviceAndroidId: string
  deviceWidth?: number
  deviceHeight?: number
  deviceBrand?: string
  deviceName?: string
  deviceModel?: string
  deviceSdkInt?: string
  deviceImei?: string
  deviceBroad?: string
  deviceBuildId?: string
  remark?: string
  createTime?: string
  updateTime?: string
  createBy?: string
  updateBy?: string
}

export interface PageParams {
  current: number
  size: number
  cardNo?: string
  deviceAndroidId?: string
  deviceBrand?: string
  deviceModel?: string
}

/**
 * 获取设备列表
 */
export function getScriptCardDeviceList(params: PageParams) {
  return request({
    url: '/script/cardDevice/list',
    method: 'get',
    params
  })
}

/**
 * 添加设备绑定
 */
export function addScriptCardDevice(data: ScriptCardDevice) {
  return request({
    url: '/script/cardDevice/add',
    method: 'post',
    data
  })
}

/**
 * 删除设备绑定
 */
export function deleteScriptCardDevice(id: string) {
  return request({
    url: `/script/cardDevice/delete/${id}`,
    method: 'delete'
  })
}

/**
 * 批量删除设备绑定
 */
export function batchDeleteScriptCardDevice(ids: string[]) {
  return request({
    url: '/script/cardDevice/batchDelete',
    method: 'delete',
    data: ids
  })
}

/**
 * 获取设备详情
 */
export function getScriptCardDeviceDetail(id: string) {
  return request({
    url: `/script/cardDevice/detail/${id}`,
    method: 'get'
  })
}

/**
 * 统计卡密已绑定的设备数量
 */
export function countBoundDevices(cardNo: string) {
  return request({
    url: `/script/cardDevice/count/${cardNo}`,
    method: 'get'
  })
}
